content/blog/architecture-as-code-serverless/index.md

```diff
@@ -0,0 +1,170 @@
+
+<!--more-->
+
+## Overview
+
+Serverless is a pattern for building services without having to manage servers. You write a service and deploy it on infrastructure managed by a cloud service provider. Serverless apps can be either a simple function invoked by an event or an application in a container. They are typically ephemeral, where the application runs within a fixed time or until it completes the task. Managed implementations such as AWS Lambda and Fargate, Azure Functions, and Google Cloud Run and Cloud Functions charge for the service only when it’s invoked and create instances, or workers, based on demand.
+
+Developers can write the application in any language that the service provider supports. That means developers can choose the language that best meets the needs of their application. For example, a compiled application may respond more quickly than a scripting language. Serverless applications can be either synchronous or asynchronous, depending on the task.
+
+Another component of serverless architecture is an API Gateway which maps HTTP operations such as GET, PUT, POST, and DELETE to functions. A gateway simplifies implementing REST interfaces for functions and applications.
+
+Let’s take a look at examples where we can apply Architecture as Code to serverless.
+
+## Reusing resources
+
+While serverless simplifies and reduces management overhead for deploying functions and containers, ancillary services also need to be configured. For example, a production deployment requires creating clusters, security groups, load balancers, and IAM policies. The following Python script creates resources that can be imported into `__main__.py`, which deploys the function or container. It ensures that each serverless application uses a vetted set of resources that can be replicated for every deployment.
+
+```python
+from pulumi import export, ResourceOptions
+import pulumi_aws as aws
+import json
+
+# Create an ECS cluster to run a container-based service.
+cluster = aws.ecs.Cluster('cluster')
+
+# Read back the default VPC and public subnets, which we will use.
+default_vpc = aws.ec2.get_vpc(default='true')
+default_vpc_subnets = aws.ec2.get_subnet_ids(vpc_id=default_vpc.id)
+
+# Create a SecurityGroup that permits HTTP ingress and unrestricted egress.
+group = aws.ec2.SecurityGroup('web-secgrp',
+	vpc_id=default_vpc.id,
+	description='Enable HTTP access',
+	ingress=[{
+		'protocol': 'tcp',
+		'from_port': 80,
+		'to_port': 80,
+		'cidr_blocks': ['0.0.0.0/0'],
+	}],
+  	egress=[{
+		'protocol': '-1',
+		'from_port': 0,
+		'to_port': 0,
+		'cidr_blocks': ['0.0.0.0/0'],
+	}]
+)
+
+# Create a load balancer to listen for HTTP traffic on port 80.
+alb = aws.lb.LoadBalancer('app-lb',
```

Comment:
It's not quite clear to me why we need a load balancer in this context

Reply:
Pulling this from the example, more to show what is possible than practical.